distributed/deploy/spec.py

```diff
@@ -36,19 +36,39 @@ class ProcessInterface:
     It should implement the methods below, like ``start`` and ``close``
     """
 
+    @property
```

Comment:
Maybe instead we should implement this on the `Cluster` superclass in `distributed/deploy/cluster.py`

Reply:
That would also affect most of the downstream projects automatically, which would be useful.

Reply:
We could, but then ssh.Worker and ssh.Scheduler would not inherit this and I believe they were the reason I did this originally. 

I think I might end up pulling this into a decorator that can be applied to classes maybe.

Reply:
Oh, my apologies.  I saw the `spec.py` filename and assumed that this was the SpecCluster class.  I see now that it is the `ProcessInterface` class instead.  Happy to retract the comment.

I'd prefer avoiding the iterator and being explicit for now if that's ok, even given the duplication.